pisces/data_sets.py

```diff
@@ -327,6 +334,440 @@ def find_data_sets(cls, root: str | Path) -> Dict[str, 'DataSetObject']:
+        self.data_set = data_set
+        self.input_features = model_input.input_features
+        self.output_feature = output_feature
+        self.output_type = output_type
+        self.psg_type = psg_type
+        self.model_input = model_input
+
+        if isinstance(model_input, ModelInput1D):
+            self.input_window_time = model_input.input_window_time
+            self.input_sampling_hz = model_input.input_sampling_hz
+            self.input_window_samples = model_input.input_window_samples
+            self.model_input_dimension = model_input.model_input_dimension
+        elif isinstance(model_input, ModelInputSpectrogram):
+            self.input_sampling_hz = model_input.input_sampling_hz
+            self.spectrogram_preprocessing_config = model_input.spectrogram_preprocessing_config
+
+    def get_labels(self, id: str, start: int, end: int,
+                   output_feature: str) -> pl.DataFrame | None:
+        data = self.data_set.get_feature_data(output_feature, id)
+        data = data.filter(data[:, 0] >= start)
+        data = data.filter(data[:, 0] <= end)
+
+        # Mask PSG data based on accelerometer data if present
+        if "accelerometer" in self.input_features:
+            accelerometer = self.data_set.get_feature_data('accelerometer', id)
+            data = mask_psg_from_accel(data, accelerometer, 
+                                       accel_sample_rate=self.input_sampling_hz)
+
+        if self.output_feature == 'psg':
+            if self.output_type == ModelOutputType.SLEEP_WAKE:
+                y = psg_to_sleep_wake(data)
+            elif self.output_type == ModelOutputType.WAKE_LIGHT_DEEP_REM:
+                N4 = self.psg_type == PSGType.HAS_N4
+                y = psg_to_WLDM(data, N4)
+            else:
+                raise ValueError(f"Output type {self.output_type} not supported")
+        else:
+            raise ValueError(f"Output feature {output_feature} not supported")
+
+        labels = pl.DataFrame({
+            'time': data[:, 0],
+            'label': y,
+        })
+
+        return labels
+
+    def get_1D_X_for_feature(self, interpolation_timestamps: np.ndarray, 
+                             epoch_times: np.ndarray, feature_times: np.ndarray, 
+                             feature_values: np.ndarray) -> np.ndarray:
+            interpolation = np.interp(interpolation_timestamps, feature_times, feature_values)
+            X_feature = []
+            for t in epoch_times:
+                t_idx = np.argmin(np.abs(interpolation_timestamps - t))
+                # Window centered around t with half `window_samples` on each side
+                window_idx_start = t_idx - self.input_window_samples // 2
+                window_idx_end = t_idx + self.input_window_samples // 2 + 1
+                window_data = interpolation[window_idx_start:window_idx_end]
+                # reshape into (1, window_size)
+                window_data = window_data.reshape(1, -1)
```

Comment:
There's also a problem here for multidimensional vectors, this reshape will be weird (though kind of interesting on it's own...)

Reply:
Since we are passing one column at a time, will this be ever be a problem we run into?